feedback/views.py

```diff
@@ -780,6 +784,65 @@ def get_form_kwargs(self):
         return super().get_form_kwargs()
 
 
+class ImportTagsView(ManageCourseMixin, FormView):
+    template_name = "feedback_tags/import_tags.html"
+    form_class = ImportTagsForm
+
+    def get_course_options(self):
+        course_options = Course.objects.using_namespace_id(self.course.namespace).exclude(id=self.course.id)
+        user = self.request.user
+        # Logic copied from feedback/views.py:297
+        if not (user.is_superuser or user.is_staff):
+            course_options = course_options.filter(id__in=self.visible_courses)
+        return course_options
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['target_course'] = self.course
+        kwargs['course_options'] = self.get_course_options()
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        course_options = self.get_course_options()
+        existing_tags = set(FeedbackTag.objects.filter(course=self.course).values_list('slug', flat=True))
+        course_option_ids = course_options.values_list('id', flat=True)
+        tags = FeedbackTag.objects.filter(course__id__in=course_option_ids)
+        tags_dict = {
+            id: [] for id in course_option_ids
+        }
+        for tag in tags:
+            tags_dict[tag.course.id].append({
+                'name': tag.name,
+                'slug': tag.slug,
+                'description': tag.description,
+                'color': tag.color,
+                'canImport': tag.slug not in existing_tags,
+            })
+        kwargs['tags_dict'] = json.dumps(tags_dict)
+        return super().get_context_data(**kwargs)
+
+
+    def form_valid(self, form):
+        imported_tags = form.copy_tags()
+        if len(imported_tags) > 0:
+            messages.success(
+                self.request,
+                format_lazy(
+                    _('Imported successfully tags: {imported_tags}.'),
```

Comment:
This still shows the imported tags only with the tag slugs, which isn't very pretty, but it probably is quite a bit more difficult to get them properly styled, and this isn't as important now with the preview. So this is just a minor cosmetic thing, which probably shouldn't be changed at this time (to prioritize time and needs).

![Screenshot from 2024-05-27 14-57-49](https://github.com/apluslms/mooc-jutut/assets/50318434/9dcec98f-cacf-4c45-98f6-48343a50f344)